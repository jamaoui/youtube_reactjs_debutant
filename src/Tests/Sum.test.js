import Sum from "./Sum";
import {render, screen, waitFor} from "@testing-library/react";
import renderer from 'react-test-renderer'

describe('unit tests', () => {

    // Mocks

    test('sum', () => {
        const expected = 5
        render(<Sum a={2} b={3}/>)
        const element = screen.getByTestId('sum')
        expect(element).toBeInTheDocument()
        expect(element).toHaveTextContent(expected)
        expect(element).not.toHaveClass('sum')
    })

    test('class', async () => {
        render(<Sum a={2} b={3}/>)
        const element = screen.getByTestId('sum')
        expect(element).not.toHaveClass('sum')

        // render , jibo
        // click , ....
        // Expect
    })
    test('snapshot',() => {
        let root = renderer.create(<Sum a={3} b={3}/>)
        expect(root.toJSON()).toMatchSnapshot()
    })
})
