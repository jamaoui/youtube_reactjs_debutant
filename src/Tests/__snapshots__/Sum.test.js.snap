// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests snapshot 1`] = `
<div
  data-testid="sum"
>
  6
</div>
`;
